kratos/geometries/geometry.h

```diff
@@ -686,52 +686,102 @@ class Geometry : public PointerVector<TPointType>
     }
 
     /**
-     * It computes the unit normal of the geometry, if possible
+     * @brief It computes the unit normal of the geometry, if possible
+     * @param rNormal The normal in the given point
+     * @param rTangentXi The tangent in the xi direction
+     * @param rTangentEta The tangent in the eta direction
      * @return The normal of the geometry
      */
-    virtual array_1d<double, 3> AreaNormal(const CoordinatesArrayType& rPointLocalCoordinates) const
+    virtual array_1d<double, 3> NormalWithTangents(
```

Comment:
You could obtain everything in this function by simply calling F(coordinates) for the geometry and then computing the normal out of its columns.

I wouldn't add a new geometry method just for this...

Reply:
If you see in detail the algorithm you will see that the way is was computed previously is exactly the same (the tangents were computed), the thing is that it just returned the normal, this method is more general and does not break current implementation

Reply:
I really don't think we should add an interface to the geometry for this...

Reply:
It just add some extra lines and computationally does the same

Reply:
I don’t see how you define a tangent without having a plane

I’m not sure if is a good idea to add this method as @RiccardoRossi mentioned